File: src/main/java/com/DavidM1A2/afraidofthedark/common/event/register/RecipeRegister.java
package com.DavidM1A2.afraidofthedark.common.event.register;

import com.DavidM1A2.afraidofthedark.common.constants.ModRecipes;
import net.minecraft.item.crafting.IRecipe;
import net.minecraftforge.event.RegistryEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.registries.IForgeRegistry;

/**
 * Class that receives the register recipe event and registers all of our custom IRecipes
 */
public class RecipeRegister
{
    /**
     * Called by forge to register any of our recipes
     *
     * @param event The event to register to
     */
    @SubscribeEvent
    public void registerRecipes(RegistryEvent.Register<IRecipe> event)
    {
        IForgeRegistry<IRecipe> registry = event.getRegistry();

        // Register each item in our item list
        registry.registerAll(ModRecipes.MOD_RECIPES);
    }
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/event/register/RecipeRegister.kt
package com.davidm1a2.afraidofthedark.common.event.register

/**
 * Class that receives the register recipe event and registers all of our custom IRecipes
 */
class RecipeRegister {
    /**
     * Called by forge to register any of our recipes
     *
     * @param event The event to register to
     */
    /*
    @SubscribeEvent
    fun registerRecipes(event: RegistryEvent.Register<IForgeRegistryEntry<IRecipe>>) {
        val registry = event.registry

        // Register each item in our item list
        registry.registerAll(*ModRecipes.MOD_RECIPES)

        RecipeSerializers.register(RecipeSerializers.SimpleSerializer("crafting_research_scroll") { ScrollCombineRecipe() })
    }
     */
}